### ISSUE TITLE
File with single interface export compilation failed

### ISSUE THREAD

**Issue by thefabulousdev** [frankwallis/plugin-typescript#117]
Version : **4.0.5**

#### Given 

We have a file named `IShape.ts` which contains
```ts
export interface IShape {
    getArea() : number;
}
```

#### When 

The file is required in a file named `Square.ts` which contains an implementation of the `IShape` interface
```ts
import {IShape} from './IShape';

export class Square extends IShape {
    constructor(protected width=0) {

    }

    getArea() : number {
        return this.width * this.width;
    }
}
```
and we run 
```bash
$ jspm build ...
``` 
#### Then
jspm does a non-zero exit with 
```bash
TypeScript Cannot find module './IShape'. (TS2307)
err  Error: Typescript compilation failed
    at file:///home/joel/Documents/Personal/open-source/typescript-jspm-seed-project/typescript-jspm-seed/jspm_packages/github/frankwallis/plugin-typescript@4.0.5/plugin.js:66:27

```

#### Finally

Is this related to #97 ?

**Comment by frankwallis** [comment]
I think this is more likely to be a file resolution issue due to it looking for ```./IShape``` instead of ```./IShape.ts```. Try adding a ```defaultExtension: 'ts'``` to your packages configuration to tell systemjs to add the ```.ts``` extension when resolving imports. See [here](https://github.com/frankwallis/plugin-typescript/blob/master/examples/angular/jspm.config.js#L12) for example.

**Comment by thefabulousdev** [comment]
Hey Frank, that is not the issue, I can load other files which export classes and are used in some way with no problem.

In my `jspm.config.js`
```js
SystemJS.config({
  transpiler: "plugin-typescript",
  packages: {
    "typescript-jspm-seed": {
      "main": "main.ts",
      "defaultExtension": "ts",
      "meta": {
        "*.ts": {
          "loader": "plugin-typescript"
        }
      }
    }
  }
});
```

I believe this might be related to the fact that TypeScript interfaces are not compiled to code ? You can try it [here](https://github.com/thefabulousdev/typescript-jspm-seed), an `npm install` and `node run build:umd` will expose you the issue.

**Comment by thefabulousdev** [comment]
Solved.

I was using 
```js
  typescriptOptions: {
    typeCheck : 'strict'
  }
```

**Comment by thefabulousdev** [comment]
Actually, the flag only allows the build to continue in despite of errors, re-opening issue.

**Comment by frankwallis** [comment]
Thank you for raising this, it was happening because (as you said) the file is not needed at runtime, so it was not getting included in the build. I have added a fix and it should be resolved in 4.0.6.

**Comment by thefabulousdev** [comment]
No problem!  Thanks 😄
